package.json

```diff
@@ -10,7 +10,8 @@
     "type": "git",
     "url": "https://github.com/aduth/hpq.git"
   },
-  "main": "dist/hpq.js",
+  "browser": "dist/hpq.js",
+  "main": "src/index.js",
```

Comment:
Because the file includes ES module import, I don't think we can point to `src/index.js` as the `main`, at least not also without also adding `"module": true` to this file and bumping the minimum required Node.js version to 13.2.0+ ([release](https://nodejs.org/en/blog/release/v13.2.0/)).

Alternatively, we could create a separate bundle generated from the rollup configuration which explicitly targets Node environments (like the existing UMD, but with `process.browser` as `false`).

Alternatively alternatively, is there anything we could use to evaluate _at runtime_ as detecting whether the environment is a browser or not? To use in place of the `process.browser` constant. My concern here would have been that I don't like the idea of having a `require` function call in browser-bundled code, but I guess that's already the case anyways with these changes (unless we did some dead code removal somehow).

Alternatively alternatively alternatively, we could have the `main` entry point be a file who's sole contents are responsible for bootstrapping the DOM and requiring the existing distributable:

```
// hpq.node.js
global.document = ( new JSDOM() ).window;
module.exports = require( './hpq.js' );
```

The downside here is that we lose any advantage against polluting the global namespace, where your changes as proposed nicely limit the use of JSDOM to `DOMImplementation`, rather than to create an entire `window` object.

Reply:
You are totally right, I missed the fact that this code uses some features which Node 10 doesn't support. I will revisit the current proposal when I find some free time. Thank you for looking at it.

> The downside here is that we lose any advantage against polluting the global namespace, where your changes as proposed nicely limit the use of JSDOM to `DOMImplementation`, rather than to create an entire `window` object.

I wanted to avoid that, because of the issues with the `jsdom` implementation. It uses some Node.js APIs which don't exist in React Native.